### PR TITLE
mkdir codemod fixes

### PR DESCRIPTION
**Pull request by clavedeluna**
#577 wasn't    quite ready to be merged, but no worries, the fixes are here!

- correctly pass leading lines
- unit test to show codemod works in function scope
- report unfixed for cases we don't yet handle
- add missing custom node is selected logic and test

### CODE REVIEW

File: src/core_codemods/tempfile_mktemp.py [hunk 1]
```diff
@@ -115,6 +159,21 @@ def _mktemp_is_sink(
...
+        self.calls = []
+
+    def visit_Call(self, node: cst.Call) -> None:
+        (
+            self.calls.append(node)
+            if self.find_base_name(node.func) == "tempfile.mktemp"
+            else None
+        )
+
+
```
Comment (andrecsilva) [inline src/core_codemods/tempfile_mktemp.py]: Is this used anywhere?

File: src/core_codemods/tempfile_mktemp.py [hunk 2]
```diff
@@ -19,25 +23,64 @@ class TempfileMktempTransformer(
...
+        context: CodemodContext,
+        results: list[Result] | None,
+        file_context: FileContext,
+        _transformer: bool = False,
+    ):
+        self.mktemp_calls: set[cst.Call] = set()
+        super().__init__(context, results, file_context, _transformer)
+
+    def visit_Call(self, node: cst.Call) -> None:
+        self.mktemp_calls.add(node) if self._is_mktemp_call(node) else None
```
Comment (andrecsilva) [inline src/core_codemods/tempfile_mktemp.py]: The `else` is useless here.

File: src/core_codemods/tempfile_mktemp.py [hunk 3]
```diff
@@ -19,25 +23,64 @@ class TempfileMktempTransformer(
...
+                        wrapper_func_name,
+                        original_node.leading_lines,
+                        assignment=False,
+                    )
+
+        # If we get here it's because there is a mktemp call but we haven't fixed it yet.
+        for unfixed in self.mktemp_calls:
+            self.report_unfixed(unfixed, reason="Pixee does not yet support this fix.")
+        self.mktemp_calls.clear()
         return updated_node
```
Comment (andrecsilva) [inline src/core_codemods/tempfile_mktemp.py:67]: This logic feels weird here.

Say a mktemp call `mcall` is contained in the statement, `leave_Call` will be called first and `mcall` will be added to `self.mktemp_calls`. 

Afterwards, `leave_simpleStatementLine` will run. If `mcall` is fixed, it will be reported as changed, but since its removal from `self.mktemp_calls` only happens at the end, it will also be reported as unfixed.

In short, can you add a test to show that no call is being reported as fixed and unfixed at the same time?

**Review by andrecsilva — CHANGES_REQUESTED** [review]
Some small concerns.

File: src/core_codemods/tempfile_mktemp.py (hunk 2 above)
Comment (clavedeluna) [reply src/core_codemods/tempfile_mktemp.py]: you can't write a ternary expression in Python without an else. Just wanted to make it one line instead of
```
if self._is_mktemp_call(node):
    self.mktemp_calls.add(node)
```

File: src/core_codemods/tempfile_mktemp.py (hunk 3 above)
Comment (clavedeluna) [reply src/core_codemods/tempfile_mktemp.py:67]: I checked thought I checked this. The key is the `return` statements above. If the code path hits the fixed cases we have handled, it returns, never reaching this code. Easy enough to add an assert statement to demonstrate, so will do.

File: src/core_codemods/tempfile_mktemp.py (hunk 1 above)
Comment (clavedeluna) [reply src/core_codemods/tempfile_mktemp.py]: woops thanks for the good eye, old code

File: src/core_codemods/tempfile_mktemp.py (hunk 3 above)
Comment (clavedeluna) [reply src/core_codemods/tempfile_mktemp.py:67]: see https://github.com/pixee/codemodder-python/pull/586/files#diff-d452b243674c71654239be30328a960ed2c1e0aea06bc5b57c1ab7c8169c3cb9R43

Comment (andrecsilva) [reply src/core_codemods/tempfile_mktemp.py:67]: > I checked thought I checked this. The key is the `return` statements above. If the code path hits the fixed cases we have handled, it returns, never reaching this code. Easy enough to add an assert statement to demonstrate, so will do.

The problem is when you have two or more mktemp calls. If the first one is fixed but the second one is not. When you report the second one, the first should still be included in `self.mktemp_calls`. This results in it being reported as unfixed.

Comment (clavedeluna) [reply src/core_codemods/tempfile_mktemp.py:67]: Example please. Because what you're describing sounds like a super edge case of two calls in the same line and we don't really handle those cases or test that for any codemod right now.

Comment (andrecsilva) [reply src/core_codemods/tempfile_mktemp.py:67]: Just include `self.mktemp_calls.clear()` before the returns in `leave_SimpleStatementLine` and I'll be satisfied.

**Review by andrecsilva — APPROVED** [review]
